dask_expr/statistics.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import annotations
```

Comment:
I could use some help understanding this module.  

I think that in general we have yet to define what kinds of statistics we're going to capture, and how we plan to encode those.  There are lots of options here.  

I _think_ what I'm seeing here is that your response is "we'll just make different classes for all the different kinds of things that people might want to encode".  Is that correct?  If so, I'm not totally bought into this just yet.  

I think that the question of "how do we encode dataframe-level or partition-level statistics" is a big open one.  I'm ok with us not having a clear answer on this before we move forward, but I want the level of sophistication of our solution to be correlated with our confidence.  This feels like a somewhat sophisticated/specific solution (a few classes with some specific method APIs) but I don't have confidence that it's correct (or at least I don't know enough to be confident).  Can you help me understand here?